packages/aws-cdk/lib/api/deployments.ts

```diff
@@ -430,7 +431,12 @@ export class Deployments {
   }
 
   public async stackExists(options: StackExistsOptions): Promise<boolean> {
-    const { stackSdk } = await this.prepareSdkFor(options.stack, undefined, Mode.ForReading);
+    let stackSdk;
+    if (options.tryLookupRole) {
+      stackSdk = (await this.prepareSdkWithLookupOrDeployRole(options.stack)).stackSdk;
+    } else {
+      stackSdk = (await this.prepareSdkFor(options.stack, undefined, Mode.ForReading)).stackSdk;
+    }
```

Comment:
`prepareSdkWithLookupOrDeployRole` defaults to `ForReading`. If we can't assume the lookup role, it will fall back to the deploy role. 

Don't use `prepareSdkFor` directly, use the wrappers like `prepareSdkWithLookupRole` or whichever one you want to use; those methods have names that make it clear which role is being assumed.

Reply:
We should still try to fallback to the deploy role, in case somebody doesn't have permissions to the lookup role. Why don't we catch any errors we get when trying to assume the deploy role? If we can't assume it then we stop changeset creation. 

Inside of `prepareSdkFor`, `cachedSdkForEnvironment` returns an object with `didAssumeRole` on it. `didAssumeRole` is not returned to the caller, but you could add it as a property to the returned object. Then you have a way of checking if the fallback was successful.

Reply:
discussed off(on?)line, we can investigate this in a follow up next week